RN v12.2.1 Preview (#1131)
mia-platform/documentation: docs/release-notes/v12.2.1.md

## Fast Data

### Fast Data Services

#### Single View Trigger Generator

The new version `v3.1.5` of the _Single View Creator_ is available!

##### Bug Fix

This version addressed a few bugs, here is a comprehensive list:

* in case of one-to-many relationships, the service does not crash anymore when it executes an automatic strategy over a record whose primary key fields have a value set to `null`

For more information check out the service [changelog](/runtime_suite/single-view-trigger-generator/changelog.md).

#### Single View Creator

In order to simplify the creation of Single View Creator plugins depending on the selected Fast Data Architecture, the original marketplace plugin has been split into two. The Mongo based plugin is  preconfigured for the [Standard architecture](/fast_data/architecture.md#standard-architecture), while the Kafka based is preconfigured for the [Event-Driven architecture](/fast_data/architecture.md#event-driven-architecture).

![Single View Creator Marketplace](./img/v12.2.1/svc-marketplace.png)

## Data Catalog

#### Data Catalog Agent Template

The new version `v0.4.0` of the _Data Catalog Agent_ template is available!

##### New Features

> - Mia-Platform CRUD Service (`v6.10.0` or later) has been added among the supported Data Sources, so that it is possible to configure the Agent to gather metadata from the ones deployed in your projects environments.

Author: ```suggestion
Mia-Platform CRUD Service (`v6.10.0` or later) has been added among the supported Data Sources, so that it is possible to configure the Agent to gather metadata from the ones deployed in your projects environments.
```

- to better clarify the error response, when requesting a document `__STATE__` to transition to an invalid value, a _400 Bad Request_ HTTP Error is returned, rather than a _404 Not Found_ HTTP Error
- querying documents with fields not specified in the collection data model now yields a _400 Bad Request_ HTTP Error to enforce data model access. Similarly, documents fields not defined in the collection data model are not returned in the response

##### New Features

- introduced `defaultSorting` as a new collection configuration property. This allows to define a set of fields with their sorting direction to be employed as sorting operation to all the GET queries whenever the `_s` parameter is not specified. In the case `_s` query parameter is set, it takes precedence over the default sorting assigned to the collection config.
By default, the property is not set, so that the default behavior with respect to previous service versions remains unchanged.

> A migration guide is available [here](/runtime_suite/crud-service/guides/Migration-Guide-v7.md) to aid you in migrating your existing CRUD services to version `v7.0.0`.

Author: ```suggestion
A migration guide is available [here](/runtime_suite/crud-service/guides/10_migration-guide-v7.md) to aid you in migrating your existing CRUD services to version `v7.0.0`.
```

> #####
> 

Author: ```suggestion
```

For self-hosted installations, please read the [following guidelines](#how-to-update-your-console).
:::

## Console

### Revision History for Projects with Enhanced Workflow

We have introduced the possibility to view the history of revisions for projects using the Enhanced Workflow.

Now, you can get a quick overview of all the saves made within a specific revision, including additional information such as the date & time, author, and the message associated with the revision save.

You can access this feature by clicking on the "History" button inside the revision action popover: a modal will show up, will all the afore-mentioned information.

### Improvements

#### Telepresence v2 support in Debug area

Support for telepresence v2 has been added within the Debug area of the console. The commands that are displayed via the appropriate modal are now supported with the latest version of the tool.

For more information [go to the documentation](/development_suite/debugging/telepresence.md).

#### `miactl` CLI - Added a flag to show all the public Markeplace Items

> A new `--public` flag has been added to the `miactl marketplace list` command: it allows to fetch and visualize not only the items of the current requested Company, but also all the public items from other Companies.
> Check the [`miactl marketplace list` command documentation](/cli/miactl/commands#list-5) for more information.

Reviewer: ```suggestion
A new `--public` flag has been added to the `miactl marketplace list` command: it allows to fetch and visualize not only the items of the current requested Company, but also all the public items from other Companies.

Check the [`miactl marketplace list` command documentation](/cli/miactl/commands.md#list-5) for more information.
```

:::

## Console

### Revision History for Projects with Enhanced Workflow

We have introduced the possibility to view the history of revisions for projects using the Enhanced Workflow.

Now, you can get a quick overview of all the saves made within a specific revision, including additional information such as the date & time, author, and the message associated with the revision save.

You can access this feature by clicking on the "History" button inside the revision action popover: a modal will show up, will all the afore-mentioned information.

### Improvements

#### Telepresence v2 support in Debug area

Support for telepresence v2 has been added within the Debug area of the console. The commands that are displayed via the appropriate modal are now supported with the latest version of the tool.

For more information [go to the documentation](/development_suite/debugging/telepresence.md).

#### `miactl` CLI - Added a flag to show all the public Markeplace Items

A new `--public` flag has been added to the `miactl marketplace list` command: it allows to fetch and visualize not only the items of the current requested Company, but also all the public items from other Companies.

> Check the [`miactl marketplace list` command documentation](/cli/miactl/commands.md#list-5) for more information.

Reviewer: ```suggestion
Check the [`miactl marketplace list` command documentation](/cli/miactl/30_commands.md#list-5) for more information.
```

- querying documents with fields not specified in the collection data model now yields a _400 Bad Request_ HTTP Error to enforce data model access. Similarly, documents fields not defined in the collection data model are not returned in the response

##### New Features

- introduced `defaultSorting` as a new collection configuration property. This allows to define a set of fields with their sorting direction to be employed as sorting operation to all the GET queries whenever the `_s` parameter is not specified. In the case `_s` query parameter is set, it takes precedence over the default sorting assigned to the collection config.
By default, the property is not set, so that the default behavior with respect to previous service versions remains unchanged.

A migration guide is available [here](/runtime_suite/crud-service/guides/10_migration-guide-v7.md) to aid you in migrating your existing CRUD services to version `v7.0.0`.

For more information please check out the service [changelog](/runtime_suite/crud-service/changelog.md).

## How to update your Console

> For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100-how-to-upgrade.md#v12---version-upgrades) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ `v12.X.X`.

Author: ```suggestion
For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100-how-to-upgrade.md#v12---version-upgrades) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ `v12.0.2-beta.0`.
```

> For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100-how-to-upgrade.md#v12---version-upgrades) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ `v12.0.2-beta.0`.

Author: ```suggestion
For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100-how-to-upgrade.md#v12---version-upgrades) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ `v12.0.3-beta.0`.
```
